### PR TITLE
Bnc940892 lanudevauto internal error

### PR DESCRIPTION
**Pull request by mchf**

### CODE REVIEW

File: src/modules/LanUdevAuto.rb [hunk 1]
```diff
@@ -184,7 +184,7 @@ def Import(settings)
           createUdevFromIfaceName(Ops.get_list(settings, "interfaces", []))
         )
       else
-        @udev_rules = Ops.get_list(settings, "net-udev", [])
+        @udev_rules = settings["net-udev"] || {}
```
Comment (ancorgs) [inline src/modules/LanUdevAuto.rb]: shouldn't this be ``settings["net-udev"] || []`. Looking at the rest of the code, it seems to be an array of hashes, not a hash.

Comment (mchf) [inline src/modules/LanUdevAuto.rb]: you're right. thx

**Comment by ancorgs** [comment]
lgtm

**Comment by kobliha** [comment]
Hmm, sounds like we actually need a test case?

**Comment by mchf** [comment]
thanks for review.

**Comment by mchf** [comment]
hups, wrong button :-)

**Comment by kobliha** [comment]
OK, thanks, feel free to merge

File: test/lan_udev_auto_test.rb [hunk 2]
```diff
@@ -105,10 +105,29 @@
...
+    udev_rules = [
+      {
+        "rule"  => ATTR,
+        "value" => VALUE,
+        "name"  => NAME
+      }
+    ]
+
+    expect(LanUdevAuto.Import({ "net-udev" => udev_rules }))
+      .to be_true
```
Comment (ancorgs) [inline test/lan_udev_auto_test.rb]: better use `.to eq true`

File: test/lan_udev_auto_test.rb [hunk 3]
```diff
@@ -105,10 +105,29 @@
...
-      LanUdevAuto.Import({ "net-udev" => udev_rules })
-      LanUdevAuto.Write
-    end
+    LanUdevAuto.Import({ "net-udev" => udev_rules })
+    LanUdevAuto.Write
+  end
+end
+
+describe "LanUdevAuto#Import" do
+  it "imports defined udev rules correctly" do
```
Comment (ancorgs) [inline test/lan_udev_auto_test.rb]: This does not test that there are imported correctly. It only tests it does not crash.

File: test/lan_udev_auto_test.rb [hunk 4]
```diff
@@ -77,11 +77,11 @@
 
 end
 
-describe "LanUdevAuto#Write" do
+ATTR = "ATTR{address}"
+VALUE = "aa:BB:cc:DD:ee:FF"
+NAME = "custom-name"
```
Comment (ancorgs) [inline test/lan_udev_auto_test.rb]: Better use `let` instead of polluting the global namespace with constants

**Comment by ancorgs** [comment]
Yes, I know it's already merged, but I couldn't resist to point some things in the test.

**Comment by mchf** [comment]
Thanks for notes. I'll fix it when merging into master

File: test/lan_udev_auto_test.rb (hunk 3 above)
Comment (mchf) [inline test/lan_udev_auto_test.rb]: yes it was targeted mainly against rewriting from Builtins into ruby, so a smoke test.

**Pull request merged**